datacube/index/postgres/_api.py

```diff
@@ -868,3 +873,124 @@ def grant_role(self, role, users):
                 raise ValueError('Unknown user %r' % user)
 
         tables.grant_role(self._connection, pg_role, users)
+
+
+    # S3 specific functions
+    # See .tables for description of each column
+    def put_s3_mapping(self, dataset_ref, band, s3_dataset_id):
```

Comment:
Still not convinced by s3 specifics polluting this already crowded class. It would be great to have these separate. If we get more drivers the problem will only become worse.